crates/starknet/types/src/lib.rs

```diff
@@ -85,3 +85,54 @@ impl From<Call> for starknet::core::types::Call {
         }
     }
 }
+
+/// Validates that a Felt value represents a valid Starknet contract address.
+///
+/// In Starknet, contract addresses must follow specific constraints to be considered valid:
+/// - They must be greater than or equal to 2, as addresses 0 and 1 are reserved for system use:
+///   * 0x0 acts as the default caller address for external calls and has no storage
+///   * 0x1 functions as a storage space for block mapping [link](https://docs.starknet.io/architecture-and-concepts/network-architecture/starknet-state/#special_addresses)
+/// - They must be less than 2^251 + 17 * 2^192 (0x800000000000000000000000000000000000000000000000000000000000000)
```

Comment:
`0x800000000000000000000000000000000000000000000000000000000000000` is just `2^251`. `+ 17 * 2^192 ` is not part of it